numpy/core/_methods.py

```diff
@@ -177,6 +179,11 @@ def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
     if dtype is None and issubclass(arr.dtype.type, (nt.integer, nt.bool_)):
         dtype = mu.dtype('f8')
 
+    # TODO: keepdims, out
+    if rcount < 2:
+        dtype = dtype or arr.dtype
+        return dtype.type('nan') if rcount == 0 else dtype.type(0)
```

Comment:
Also has the wrong shape. Can you avoid this special case? What happens if you just run the code below?

Reply:
To elaborate - currently (and correctly):
```python
>>> np.var(np.eye(3), axis=())
array([[ 0.,  0.,  0.],
       [ 0.,  0.,  0.],
       [ 0.,  0.,  0.]])
```

With your patch, this returns `0.0`.

Reply:
I didn't think of that. Anyway I needed some form of special case, since `sumsquarediff` always operate on one axes